app/endpoints/recommendation.py

```diff
@@ -0,0 +1,176 @@
+from fastapi import APIRouter, Depends, File, HTTPException, UploadFile
+from fastapi.responses import FileResponse
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from app.core.config import Settings
+from app.cruds import cruds_recommendation
+from app.dependencies import (
+    get_db,
+    get_request_id,
+    get_settings,
+    is_user_a_member,
+    is_user_a_member_of,
+)
+from app.models import models_core
+from app.schemas import schemas_recommendation
+from app.utils.tools import get_file_from_data, save_file_as_data
+from app.utils.types import standard_responses
+from app.utils.types.groups_type import GroupType
+from app.utils.types.tags import Tags
+
+router = APIRouter()
+
+
+@router.get(
+    "/recommendation/recommendations",
+    response_model=list[schemas_recommendation.Recommendation],
+    status_code=200,
+    tags=[Tags.recommendation],
+)
+async def get_recommendation(
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member),
+):
+    """
+    Get recommendations.
+
+    **The user must be authenticated to use this endpoint**
+    """
+
+    return await cruds_recommendation.get_recommendation(db=db)
+
+
+@router.post(
+    "/recommendation/recommendations",
+    response_model=schemas_recommendation.Recommendation,
+    status_code=201,
+    tags=[Tags.recommendation],
+)
+async def create_recommendation(
+    recommendation: schemas_recommendation.RecommendationBase,
+    db: AsyncSession = Depends(get_db),
+    settings: Settings = Depends(get_settings),
+    user: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.BDE)),
+):
+    """
+    Create a recommendation.
+
+    **This endpoint is only usable by members of the group BDE**
+    """
+
+    try:
+        return await cruds_recommendation.create_recommendation(
+            recommendation=recommendation,
+            db=db,
+            settings=settings,
+        )
+    except ValueError as error:
+        raise HTTPException(status_code=422, detail=str(error))
```

Comment:
```suggestion
        raise HTTPException(status_code=400, detail=str(error))
```

Reply:
In others modules, the status code returned for `ValueError` is often also 422. Should I create a PR where I replace all these status code by 400 ?